### PR TITLE
Spearman to semeioscript

### LINKED ISSUE #24: Spearman job output/feedback
The reporter should be utilised to publish the data. In particular, the following should be made available:

- the correlation matrix (previously #48)
- the clusters (this is currently outputted)

### PR DESCRIPTION
**Pull request by ManInFez**
resolves #24

### CODE REVIEW

File: tests/jobs/spearman_correlation_job/test_integration.py [hunk 1]
```diff
@@ -38,6 +38,20 @@ def test_main_entry_point_gen_data(monkeypatch):
     # again is a tuple containing the configuration which is a list of configs.
     assert len(run_mock.call_args[0][0]) == 47, "wrong number of clusters"
 
+    cor_matrix_file = (
+        "storage/snake_oil/ensemble/reports/"
+        "SpearmanCorrelationJob/correlation_matrix.json"
+    )
+    with open(cor_matrix_file) as f:
+        json.load(f)
```
Comment (markusdregi) [inline tests/jobs/spearman_correlation_job/test_integration.py:46]: Can we do any testing besides being valid JSON?

**Comment by markusdregi** [comment]
Besides the question of testing the content of the correlation matrix this looks really good 👍 If I can have a rebase on top of master at the same time that would be appreciated...

File: tests/jobs/spearman_correlation_job/test_integration.py (hunk 1 above)
Comment (ManInFez) [reply tests/jobs/spearman_correlation_job/test_integration.py:46]: Do you know any meaningful tests to the matrix? I am unsure how to calculate it by hand.

File: semeio/jobs/correlated_observations_scaling/job.py [hunk 2]
```diff
@@ -32,9 +35,21 @@ def scale(self, measured_data):
         measured_data.filter_ensemble_std(config.CALCULATE_KEYS.std_cutoff)
 
         matrix = DataMatrix(measured_data.data)
-        matrix.std_normalization(inplace=True)
+        matrix.normalize_by_std()
```
Comment (oyvindeide) [inline semeio/jobs/correlated_observations_scaling/job.py:38]: This behavior is now changed to returning a copy, but that copy is never stored. That should have made the tests fail, but apparently does not (should be fixed?)

File: semeio/communication/reporter.py [hunk 3]
```diff
@@ -10,9 +11,17 @@ def __init__(self, output_dir):
             err_fmt = "Expected output_dir to be an absolute path, received '{}'"
             raise ValueError(err_fmt.format(output_dir))
 
+    def publish_csv(self, namespace, data):
+        output_file = self._prepare_output_file(namespace) + ".csv"
+        if isinstance(data, pd.DataFrame):
```
Comment (markusdregi) [inline semeio/communication/reporter.py:16]: The typical Python approach here is to support publishing as csv if it has a `to_csv`-method. Duck typing 🦆

Comment (ManInFez) [reply semeio/communication/reporter.py]: 🦆

File: semeio/jobs/scripts/misfit_preprocessor.py [hunk 4]
```diff
@@ -11,14 +11,15 @@
 from semeio.jobs.correlated_observations_scaling.exceptions import EmptyDatasetException
 
 
-class MisfitPreprocessorJob(ErtScript):  # pylint: disable=too-few-public-methods
+class MisfitPreprocessorJob(SemeioScript):  # pylint: disable=too-few-public-methods
```
Comment (jondequinor) [inline semeio/jobs/scripts/misfit_preprocessor.py:14]: is this pylint comment necessary?

File: semeio/jobs/scripts/spearman_correlation.py [hunk 5]
```diff
@@ -2,28 +2,29 @@
...
+from semeio.communication import SemeioScript
 from semeio.jobs.correlated_observations_scaling.exceptions import EmptyDatasetException
 from semeio.jobs.scripts.correlated_observations_scaling import (
     CorrelatedObservationsScalingJob,
 )
 from semeio.jobs.spearman_correlation_job.job import spearman_job
 
 
-class SpearmanCorrelationJob(ErtScript):  # pylint: disable=too-few-public-methods
+class SpearmanCorrelationJob(SemeioScript):  # pylint: disable=too-few-public-methods
```
Comment (jondequinor) [inline semeio/jobs/scripts/spearman_correlation.py:13]: is this pylint comment necessary?

File: semeio/jobs/scripts/misfit_preprocessor.py (hunk 4 above)
Comment (markusdregi) [reply semeio/jobs/scripts/misfit_preprocessor.py:14]: Yeah, pylint will complain about too few public methods in the class if the disable statement is not added 😢 We could of course turn it of globally as well, but 🤷

**Pull request merged**